src/main/java/com/thealgorithms/maths/FibonacciNumber.java

```diff
@@ -0,0 +1,42 @@
+// Created by Pronay Debnath
+// Date:- 1/10/2003
+// Updated FibonnaciNumber.java using golden ratio formula
+// Explanation:- https://www.quickanddirtytips.com/articles/what-is-the-golden-ratio-and-how-is-it-related-to-the-fibonacci-sequence/
+
+import java.util.*;
+
+public class FibonacciNumber<T extends Number> {
+
+    // Function to find the nth Fibonacci number using the golden ratio formula
+    public static <T extends Number> int nthFibonacci(T n) {
```

Comment:
Why did you use generic type here? And why `n` is of the type `T`?

Reply:
I actually used generic type for the flexibility of taking input from any numeric type that the nthFibonacci method can accept, and the second one is because it is intended to accept an input value of any numeric type. I can pass in values of different numeric types (e.g., Integer, Double) without the need for method overloading.

Reply:
Why would you sent `double` as an argument? Fibonacci sequence is a _sequence_.

Reply:
Oh I have to check that once again 😅

Reply:
Resolved the `double` argument

Reply:
Honestly I do not see the point of having this generic here. Especially for the input. I would simply expect and `int`.

Reply:
Yeah I removed the generic one